src/script.js

```diff
@@ -223,14 +223,32 @@ function populateGroups(groups) {
     }
     // clear previous images
     photoContainer.innerHTML = '';
-    groups[key].sort(function(a, b){
-      return parseInt(b.seniority) - parseInt(a.seniority)})
-               .slice(0, 8)
-               .forEach(function(MoC) {
-                 if (MoC.govtrack_id){
-                   photoContainer.innerHTML += '<img src="//www.govtrack.us/static/legislator-photos/' + MoC.govtrack_id + '-50px.jpeg" />';
-                 }
-    });
+    // randomize MoCs
+    let currIndex = groups[key].length;
+    let tempValue, randIndex;
+    while (0 !== currIndex) {
```

Comment:
Is there a reason to go through the list backwards? Also since we only need at most 8 images, while not just generate 8 random numbers and use those as the indices?

Reply:
No reason to go backwards. Generating 8 random numbers was my first idea, but I wanted to be able to handle 404 images (right now there are 3 in the 26 not-on-record group, so they showed up frequently). So, it could have been 12 or 16 random numbers to have some extras, but we'd also want to eliminate duplicates in the number array, so we don't see the same image more than once. Not hard to code, but when I thought about the potential extra iterations of checking the number array each time a new number is added, the performance advantage seemed less compelling & I guess overall it just seemed simpler to do it the way I did. Not a problem to refactor if you'd like to see it done with a random number list, though, it probably would be a little faster.

Reply:
Yeah it's a bit hard to read for what it's supposed to be doing. 
You dont need to check for dupes if you splice the array each time

this is what i was thinking (pseudocode:)
```
let mocs = [...groups[key]];  // copied so we're not mutating the array itself, I dont know if it's being used other places
let photoGroupCount = 0;
while photoGroupCount < 8: 
    let total = mocs.length;
    let index = pick random number between 0 and `total`
    let moc = mocs.splice(index, 1) // mutates the original array to remove one object
    let success = tryToGetAndAppendImage(moc) // might need to be an async/await situation or recursive call like you did
    if (success) {
        photoGroupCount++;
    }
```